### ISSUE TITLE
Issue updating a shared contact

### ISSUE THREAD

**Issue by krispayne** [hanneshayashi/gsm#9]
When updating  a shared contact, stdout is empty, the return code is 1, and the log shows "Error creating shared contact: EOF"

The command used is: 
```gsm sharedContacts update --email "address=jane@doe.net;primary=true;label=work;displayName=Jane Doe" --url "https://www.google.com/m8/feeds/contacts/example.com/base/exampleIDCode"```

Expected result: a contact is updated.

As far as I can tell, I have enough required flags in the command. I have no issue listing, getting, or creating a contact.

**Comment by krispayne** [comment]
I also notice that when creating a new shared contact, the email does not get added. Maybe this is why my updates are failing, since they are attempting to add email addresses:
```[kris@gemini ~]% gsm sharedContacts create --domain "example.com" --givenName "Jane" --familyName "Doe" --email "displayName=Jane Doe;address=jane@doe.net;primary=true" --phoneNumber "phoneNumber=+1 800 12345678;primary=true;label=Work" --organization "orgName=Jane Doe's Co"
<Entry gd="http://schemas.google.com/g/2005" xmlns="http://www.w3.org/2005/Atom">
	<where rel="" valueString="" href="">
		<entry>
			<id></id>
			<category scheme="" term=""></category>
			<content></content>
			<link rel="" type="" href=""></link>
			<postalAddress></postalAddress>
			<geoPt lat="" lon=""></geoPt>
			<phoneNumber></phoneNumber>
			<email address=""></email>
		</entry>
	</where>
	<name>
		<givenName>Jane</givenName>
		<additionalName></additionalName>
		<familyName>Doe</familyName>
		<namePrefix></namePrefix>
		<nameSuffix></nameSuffix>
		<fullName>Jane Doe</fullName>
	</name>
	<category scheme="http://schemas.google.com/g/2005#kind" term="http://schemas.google.com/contact/2008#contact"></category>
	<title type="">Jane Doe</title>
	<id>http://www.google.com/m8/feeds/contacts/example.com/base/762c046a89bcb906</id>
	<updated>2022-08-22T22:59:15.536Z</updated>
	<content></content>
	<link rel="http://schemas.google.com/contacts/2008/rel#photo" type="image/*" href="https://www.google.com/m8/feeds/photos/media/example.com/762c046a89bcb906?v=3.0"></link>
	<link rel="self" type="application/atom+xml" href="https://www.google.com/m8/feeds/contacts/example.com/full/762c046a89bcb906?v=3.0"></link>
	<link rel="edit" type="application/atom+xml" href="https://www.google.com/m8/feeds/contacts/example.com/full/762c046a89bcb906/1661209155536413?v=3.0"></link>
	<organization label="" primary="" rel="http://schemas.google.com/g/2005#other">
		<orgDepartment></orgDepartment>
		<orgJobDescription></orgJobDescription>
		<orgName>Jane Doe&#39;s Co</orgName>
		<orgSymbol></orgSymbol>
		<orgTitle></orgTitle>
		<where></where>
	</organization>
	<phoneNumber rel="" primary="true" label="Work" uri="">+1 800 12345678</phoneNumber>
</Entry>%```

**Comment by hanneshayashi** [comment]
Thanks for the report! I can confirm that there seems to be an issue with updating shared contacts and it seems like it is related to a missing header. Not sure what changed, but I will take a look.
The fix will most likely make it into the next version of GSM (which I am currently working on), but it may take a week or two. Sorry about that.

**Comment by krispayne** [comment]
No problem, only just started testing so not in a hurry 😊

**Comment by hanneshayashi** [comment]
Should hopefully be fixed in [GSM v0.8.0](https://github.com/hanneshayashi/gsm/releases/tag/v0.8.0). Please try it and feel free to close this issue if it works (or report back if it doesn't)

**Comment by krispayne** [comment]
Seems to be working!
